python/ml_wrappers/model/image_model_wrapper.py

```diff
@@ -372,15 +420,96 @@ def predict_proba(self, dataset, iou_threshold=0.1):
 
         :param dataset: The dataset to predict_proba on.
         :type dataset: ml_wrappers.DatasetWrapper
-        param iou_threshold: amount of acceptable error.
+        :param iou_threshold: amount of acceptable error.
             objects with error scores higher than the threshold will be removed
-        type iou_threshold: float
+        :type iou_threshold: float
         """
         predictions = self.predict(dataset, iou_threshold)
         prob_scores = [[pred.class_scores for pred in image_prediction] for image_prediction in predictions]
         return prob_scores
 
 
+class WrappedMlflowAutomlObjectDetectionModel:
+    """A class for wrapping an AutoML for images MLflow object detection model in the scikit-learn style."""
+
+    def __init__(self, model: PyFuncModel, classes: Union[list, np.ndarray]) -> None:
+        """Initialize the WrappedMlflowAutomlObjectDetectionModel.
+
+        :param model: mlflow model
+        :type model: mlflow.pyfunc.PyFuncModel
+        :param classes: list of class names
+        :type classes: list or np.ndarray
+        """
+
+        self._model = model
+        self._classes = classes
+        self._label_dict = {label: (i+1) for i, label in enumerate(classes)}
+
+    def _mlflow_predict(self, dataset: pd.DataFrame) -> pd.DataFrame:
+        """Perform the inference using the wrapped MLflow model.
+
+        :param dataset: The dataset to predict on.
+        :type dataset: pandas.DataFrame
+        :return: The predicted data.
+        :rtype: pandas.DataFrame
+        """
+        predictions = self._model.predict(dataset)
+        return predictions
+
+    def predict(self, dataset: pd.DataFrame, iou_thresh: float = 0.5, score_thresh: float = 0.5):
+        """Create a list of detection records from the image predictions.
+
+        :param dataset: The dataset to predict on.
+        :type dataset: pandas.DataFrame
+        :param iou_thresh: iou_threshold for nms - amount of acceptable error.
+            objects with error scores higher than the threshold will be removed
+        :type iou_thresh: float
+        :param score_thresh: Score threshold to filter by
+        :type score_thresh: float
+        :return: The predicted values.
+        :rtype: numpy.ndarray
+        """
+        image_sizes = dataset['image_size']
+
+        dataset = dataset.drop(['image_size'], axis = 1)
```

Comment:
```suggestion
        dataset = dataset.drop(['image_size'], axis=1)
```
Nit: Convention I believe